util/chunk/recordbatch.go

```diff
@@ -13,12 +13,41 @@
 
 package chunk
 
+// UnspecifiedNumRows represents requiredRows is not specified.
+const UnspecifiedNumRows = -1
+
 // RecordBatch is input parameter of Executor.Next` method.
 type RecordBatch struct {
 	*Chunk
+
+	// requiredRows indicates how many rows is considered full for parent executor.
+	// Child executor can return immediately if there are such number of rows,
+	// instead of fulling the whole chunk.
+	// This is not compulsory, so the number of returned rows can be larger than it in some cases.
+	requiredRows int
 }
 
 // NewRecordBatch is used to construct a RecordBatch.
 func NewRecordBatch(chk *Chunk) *RecordBatch {
-	return &RecordBatch{chk}
+	return &RecordBatch{chk, UnspecifiedNumRows}
+}
+
+// SetRequiredRows sets the number of rows the parent executor want.
+func (rb *RecordBatch) SetRequiredRows(numRows int) *RecordBatch {
```

Comment:
If we allocate a `RecordBatch` with zero default values, `rb := &RecordBatch{}`, and without calling `SetRequiredRows()`, 0 is not treated like unspecified.

Reply:
`UnspecifiedNumRows` has been updated to zero value of `int`.